client/src/test/java/io/pravega/client/stream/impl/SegmentReaderTest.java

```diff
@@ -78,18 +79,36 @@ public void read() throws EndOfSegmentException {
         sendData("3", outputStream);
         SegmentReader<String> segmentReader = new SegmentReaderImpl<>(factory, segment, stringSerializer, 0,
                 ClientConfig.builder().build(), controller, factory);
+
+        checkSnapshot(segment, 0, false, segmentReader.getSnapshot());
+
         String event = segmentReader.read(timeout);
         assertEquals("1", event);
+
+        checkSnapshot(segment, 16, false, segmentReader.getSnapshot());
+
         event = segmentReader.read(timeout);
         assertEquals("2", event);
+
+        checkSnapshot(segment, 32, false, segmentReader.getSnapshot());
+
         event = segmentReader.read(timeout);
         assertEquals("3", event);
 
         assertThrows("Read event", EndOfSegmentException.class, () -> segmentReader.read(timeout));
 
+        checkSnapshot(segment, 48, true, segmentReader.getSnapshot());
+
```

Comment:
Seems like segment is not sealed here, still we are asserting endOfSegment as true.

Reply:
Bascially in this test, we are using MockSegmentIoStreams. If readers read all the data than it will return endOfSegment as true [ref](https://github.com/pravega/pravega/pull/7369/files/25defd49a8a20511b5807ef0f4699960d611cbb2#diff-af0fb9dc10bb4b7ca7eca550691423261bc1aa5e0b20fccccdc59f687c6fffeeR266-R269)